components/engine/src/peekable.rs

```diff
@@ -2,13 +2,27 @@
 
 pub use crate::rocks::DBVector;
 use crate::Result;
+use keys::{PhysicalKey, ToPhysicalKeySlice};
 
 // TODO: refactor this trait into rocksdb trait.
 pub trait Peekable {
-    fn get_value(&self, key: &[u8]) -> Result<Option<DBVector>>;
-    fn get_value_cf(&self, cf: &str, key: &[u8]) -> Result<Option<DBVector>>;
+    type Key: PhysicalKey;
 
-    fn get_msg<M: protobuf::Message + Default>(&self, key: &[u8]) -> Result<Option<M>> {
+    fn get_value(
+        &self,
+        key: impl ToPhysicalKeySlice<<Self::Key as PhysicalKey>::Slice>,
```

Comment:
This syntax is ver verbose. Is there any chance we can simplify it, like use `Self::Key::Slice`? (currently it is not supported..) @kennytm @nrc @brson

Reply:
@breeswish This is rust-lang/rust#38078. I don't think this is gonna be solved soon. If you don't like the FQN here you could define a type alias elsewhere

```rust
type SliceOf<S> = <S::Key as PhysicalKey>::Slice;

...

key: impl ToPhysicalKeySlice<SliceOf<Self>>,
```